sync-mixins.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+TMPDIR="./tmp"
+MIXIN_REPO="git@github.com:giantswarm/giantswarm-kubernetes-mixin.git"
+# clone a branch if provided
+BRANCH=${1:-""}
+
+# make a temporary dir to work in
+mkdir ${TMPDIR}
```

Comment:
Instead of this `TMPDIR` I would recommend using `mktemp` in combination with trap to clean things up.

```suggestion
# make a temporary dir to work in
TMPDIR=$(mktemp --directory)
trap "rm -f $TMPDIR" EXIT
MIXIN_REPO="git@github.com:giantswarm/giantswarm-kubernetes-mixin.git"
# clone a branch if provided
BRANCH=${1:-""}
```